python/lsst/obs/base/gen2to3/translators.py

```diff
@@ -369,6 +369,18 @@ def _addDefaultRules(self):
         # Translate Gen2 str patch IDs to Gen3 sequential integers.
         self.addRule(PatchKeyHandler(), gen2keys=("patch",))
 
+        # Translate any "filter" values that appears alongside "tract" to
+        # "abstract_filter".  This is _not_ the right choice for instruments
+        # that use "physical_filter" values for coadds in Gen2 (like HSC);
+        # those will need to add a rule that invokes
+        # PhysicalToAbstractFilterKey instead for just that instrument, but the
+        # same criterial otherwise.  That will override this one, because
+        # instrument-specific rules match first, and that rule will consume
+        # the Gen2 "filter" key before this rule has a chance to fire.
+        self.addRule(CopyKeyHandler("abstract_filter", "filter"),
+                     gen2keys=("filter", "tract"),
```

Comment:
Is the goal here to process "filter" if and only if both "filter" and "tract" are in the path template? I don't really have any context for what a "rule" is.

Reply:
Yes, exactly.